shelley/chain-and-ledger/executable-spec/src/BaseTypes.hs

```diff
@@ -59,7 +60,7 @@ interval1 = UnitInterval 1
 -- | Tree like structure to represent nonce values and to support the binary
 -- operation on values.
 data Seed
-  = Nonce Integer
+  = Nonce Natural
```

Comment:
I think this is fine. There is no reason to have negative numbers here. In the implementation it would become a finite unsigned value. Would that already be possible to do now or is there a requirement for this to be unbounded?

Reply:
how about we use `Hash` here? that would allow us to use `hash(concat x y)` as seed operation (which I think is realistic)